### ISSUE TITLE
completed-docs does not report any errors

### ISSUE THREAD

**Issue by Rukenshia** [palantir/tslint#2291]
### Bug Report

- __TSLint version__: `4.5.1`
- __TypeScript version__: `2.0.2`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export function x() {
    return 1;
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "completed-docs": true
  }
}
```

#### Actual behavior
```
$ node_modules/.bin/tslint --type-check src/test.ts
Warning: The 'completed-docs' rule requires type checking
```

#### Expected behavior
Shouldn't I get a message about documentation missing? Sadly https://palantir.github.io/tslint/rules/completed-docs/ just displays a white page for me. Maybe I'm just using or imagining it wrong but ideally, I'd like to enforce everything being documented.

**Comment by ajafff** [comment]
For `--type-check` to work you also need to pass `--project ./path/to/tsconfig.json` as stated in the CLI documentation. Also the file you want to lint needs to be part of you project (included in `tsconfig.json`)

But maybe that should also be printed as error or warning ...

**Comment by Rukenshia** [comment]
Okay, that solved it. In the actual script I am running I just passed `-p tsconfig.json`, but tslint needs `--project`. I assumed I could pass it as `-p` like with tsc.

**Comment by adibbleWLT** [comment]
I am having a similar issue using tslint with Angular 4, but the above does not resolve this for me. It was working fine with Angular 2, but after upgrading this one rule does not get respected. All other lint settings, as far as currently noted, run as expected. Setting the rule as: "completed-docs": [ true, "classes", "methods", "functions" ], or even "completed-docs": true, does not cause a lint error even when I delete all documentation comments from all classes/methods/functions in a file. The file is definitely being linted as other rules do get flagged if they fail there. 

Note, the linting configuration settings did not change in the upgrade process. One thing that did change is that --project tsconfig.json is no longer passed to the command (and trying to do so gets an error that "The option '--project' is not registered with the lint command."), but instead in the "lint" section of the .angular-cli.json file. This is correctly set however, and I can change settings in the tsconfig file to affirm that the changes are being reflected in the linting output. The lint command being run is ""lint": "ng lint --type-check".
